hack/getting_started.sh

```diff
@@ -0,0 +1,551 @@
+    c=$((num>>8&0xff))
+    d=$((num&0xff))
+    echo "$a.$b.$c.$d"
+}  
+
+function binary2IP() {
+    num=`echo "ibase=2;  $1" | bc`
+    returnValue=$(num2IP $num)
+    echo ${returnValue}
+}
+
+
+function calcControlPlaneIP() {
+    subNet=`docker network inspect kind | jq -r 'map(.IPAM.Config[].Subnet) []'`
+    occupiedIPList=`docker network inspect kind | jq -r 'map(.Containers[].IPv4Address) []'`
+
+    for line in ${subNet}
+    do
+        echo ${line} | grep ":"
+        if [ $? -ne 0 ]; then
+            IPMask=${line}
+            break
+        fi
+    done
+
+    echo "IPMask is ${subNet}"
+    echo "occupiedIPList:${occupiedIPList}"
+
+    ip=`echo $IPMask | sed -En 's/^(.*)\/([0-9]{1,2})/\1/p'`
+    ipSubNetBit=`echo $IPMask | sed -En 's/^(.*)\/([0-9]{1,2})/\2/p'`
+    ipSubHostBit=$[32-${ipSubNetBit}]
+
+    IFS=.
+    hexIP=`for str in ${ip}; do printf "%02X" $str; done`
+    binaryIP=`echo "ibase=16; obase=2; ${hexIP}" | bc`
+
+    ipSubNet=`echo ${binaryIP:0:${ipSubNetBit}}`
+
+    full0="00000000000000000000000000000010"
+    full1="11111111111111111111111111111110"
+
+    minIPBinary=${ipSubNet}`echo ${full0:${ipSubNetBit}}`
+    maxIPBinary=${ipSubNet}`echo ${full1:${ipSubNetBit}}`
+
+    minIPint=`echo "ibase=2; ${minIPBinary}"|bc`
+    maxIPint=`echo "ibase=2; ${maxIPBinary}"|bc`
+
+
+    for((i=${minIPint};i<=${maxIPint};i++));  
+    do
+        ip=$(num2IP ${i})
+
+        isIPOccupied "${ip}" "${occupiedIPList}"
+        if [ $? == 0 ]; then
+            CONTROL_PLANE_ENDPOINT_IP=${ip}
+            echo "Available IP is ${CONTROL_PLANE_ENDPOINT_IP}"
+            return
+        fi
+    done
+}
+
+function configClusterctl() {
+    # Write clusterctl.yaml
+    writeByoh=0
+    clusterCtlYamlFile="${HOME}/.cluster-api/clusterctl.yaml"
+    if [ ! -f "${clusterCtlYamlFile}" ]; then
+        writeByoh=1
+        touch ${clusterCtlYamlFile}
+    else
+        grep -q "byoh" ${clusterCtlYamlFile}
+        if [ $? -ne 0 ] ; then
+            writeByoh=1
+        fi
+    fi 
+
+    if [ ${writeByoh} -eq 1 ] ; then
+        cat>>${clusterCtlYamlFile}<<EOF
+providers:
+  - name: byoh
+    url: https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/releases/latest/infrastructure-components.yaml
+    type: InfrastructureProvider
+EOF
+    fi
+
+    clusterctl config repositories | grep -q byoh
+    if [ $? -ne 0 ] ; then
+        echo "Config clusterctl failed..."
+        exit
+    fi
+}
+
+function runByohAgent(){
+    index=$1
+    byohBinaryFile=$2
+    manageClusterConfFile=$3
+
+    runCmd "docker cp ${byohBinaryFile} host${index}:/byoh-hostagent" 0 "Copying agent binary to byoh container: host${index}..."
+    runCmd "docker cp ${manageClusterConfFile} host${index}:/management-cluster.conf" 0 "Copying kubeconfig to byoh container: host${index}..."
+    docker exec -d host$i sh -c "chmod +x /byoh-hostagent && /byoh-hostagent --kubeconfig management-cluster.conf --skip-installation > /agent.log 2>&1"
```

Comment:
`--skip-installation` skips the installer part. We should not start the agent with this arg.

Reply:
We use byoh-dev/node:v1.22.3 image, don't need install anything which is mentioned in getting_started.md.

Reply:
Looks like the guide needs to be updated. Please remove `--skip-installation` from the script and switch to non-dev image (the one that is used in e2e)